File: slicing-android/app/src/main/java/com/crescentflare/slicinggame/infrastructure/physics/PhysicsBoundary.kt
package com.crescentflare.slicinggame.infrastructure.physics

import android.graphics.PointF
import android.graphics.RectF
import com.crescentflare.slicinggame.infrastructure.geometry.Vector

/**
 * Physics: a solid boundary surrounding the physics engine's space
 */
class PhysicsBoundary(override var x: Float, override var y: Float, var width: Float, var height: Float, var rotation: Float = 0f): PhysicsObject {

    // --
    // Members
    // --

    override var recursiveCheck = 0


    // --
    // Collision properties
    // --

    override val collisionBounds: RectF
        get() = RectF(0f, 0f, width, height)

    override val collisionRotation: Float
        get() = rotation

    override val collisionPivot: PointF
        get() = PointF(width / 2, height / 2)


    // --
    // Physics
    // --

    override fun onCollision(hitObject: PhysicsObject?, normal: Vector, timeRemaining: Float, physics: Physics) {
        // No implementation
    }

}

File: slicing-ios/SlicingGame/Infrastructure/Physics/PhysicsBoundary.swift
//
//  PhysicsBoundary.swift
//  Physics: a solid boundary surrounding the physics engine's space
//

import UIKit

class PhysicsBoundary: PhysicsObject {
    
    // --
    // MARK: Members
    // --

    var x: Float = 0
    var y: Float = 0
    var width: Float = 1
    var height: Float = 1
    var rotation: Float = 0
    var lethal = false
    var recursiveCheck = 0


    // --
    // MARK: Initialization
    // --
    
    init(x: Float, y: Float, width: Float, height: Float, rotation: Float = 0) {
        self.x = x
        self.y = y
        self.width = width
        self.height = height
        self.rotation = rotation
    }


    // --
    // MARK: Collision properties
    // --

    var collisionBounds: CGRect {
        get {
            return CGRect(x: 0, y: 0, width: CGFloat(width), height: CGFloat(height))
        }
    }
    
    var collisionRotation: Float {
        get {
            return rotation
        }
    }
    
    var collisionPivot: CGPoint {
        get {
            return CGPoint(x: CGFloat(width / 2), y: CGFloat(height / 2))
        }
    }


    // --
    // MARK: Physics
    // --

    func didCollide(withObject: PhysicsObject?, normal: Vector, timeRemaining: TimeInterval, physics: Physics) {
        // No implementation
    }

}
